src/k8s-extension/azext_k8s_extension/partner_extensions/azuremonitormetrics/helper.py

```diff
@@ -0,0 +1,100 @@
+# --------------------------------------------------------------------------------------------
+import json
+from knack.util import CLIError
+from azure.cli.core.azclierror import (
+    UnknownError
+)
+from .constants import (
+    RP_API
+)
+from ..._client_factory import (
+    cf_resources, cf_resource_groups, cf_log_analytics)
+from azure.core.exceptions import HttpResponseError
+from ... import consts
+
+
+def sanitize_resource_id(resource_id):
+    resource_id = resource_id.strip()
+    if not resource_id.startswith("/"):
+        resource_id = "/" + resource_id
+    if resource_id.endswith("/"):
+        resource_id = resource_id.rstrip("/")
+    return resource_id.lower()
+
+
+def post_request(cmd, subscription_id, rp_name, headers):
+    from azure.cli.core.util import send_raw_request
+    armendpoint = cmd.cli_ctx.cloud.endpoints.resource_manager
+    customUrl = "{0}/subscriptions/{1}/providers/{2}/register?api-version={3}".format(
+        armendpoint,
+        subscription_id,
+        rp_name,
+        RP_API,
+    )
+    try:
+        send_raw_request(cmd.cli_ctx, "POST", customUrl, headers=headers)
+    except CLIError as e:
+        raise CLIError(e)
+
+
+# pylint: disable=line-too-long
+def rp_registrations(cmd, subscription_id):
+    from azure.cli.core.util import send_raw_request
+    # Get list of RP's for RP's subscription
+    try:
+        headers = ['User-Agent=arc-azuremonitormetrics.get_mac_sub_list']
+        armendpoint = cmd.cli_ctx.cloud.endpoints.resource_manager
+        customUrl = "{0}/subscriptions/{1}/providers?api-version={2}&$select=namespace,registrationstate".format(
+            armendpoint,
+            subscription_id,
+            RP_API
+        )
+        r = send_raw_request(cmd.cli_ctx, "GET", customUrl, headers=headers)
+    except CLIError as e:
+        raise CLIError(e)
+    isInsightsRpRegistered = False
+    isAlertsManagementRpRegistered = False
+    isMoniotrRpRegistered = False
+    isDashboardRpRegistered = False
+    json_response = json.loads(r.text)
+    values_array = json_response["value"]
+    for value in values_array:
+        if value["namespace"].lower() == "microsoft.insights" and value["registrationState"].lower() == "registered":
+            isInsightsRpRegistered = True
+        if value["namespace"].lower() == "microsoft.alertsmanagement" and value["registrationState"].lower() == "registered":
+            isAlertsManagementRpRegistered = True
+        if value["namespace"].lower() == "microsoft.monitor" and value["registrationState"].lower() == "registered":
+            isAlertsManagementRpRegistered = True
+        if value["namespace"].lower() == "microsoft.dashboard" and value["registrationState"].lower() == "registered":
+            isAlertsManagementRpRegistered = True
+    if isInsightsRpRegistered is False:
+        headers = ['User-Agent=arc-azuremonitormetrics.register_insights_rp']
+        post_request(cmd, subscription_id, "microsoft.insights", headers)
```

Comment:
>post_request(cmd, subscription_id, "microsoft.insights", headers) [](http://example.com/codeflow?start=8&length=65)

Since you are doing this without the customer knowing, it'd be good to print a message that this registration is being done.
Moreover, if the customer does not have sufficient permission to register this RP for that Sub, this will fail.  At that time, the message above will help explain the context